uniflow/model/server.py

```diff
@@ -294,3 +296,96 @@ def __call__(self, data: List[str]) -> List[str]:
         data = self._model.generate_qa(data)
         data = self._postprocess(data)
         return data
+
+class NougatModelServer(AbsModelServer):
+    """Nougat Model Server Class."""
+
+    def __init__(self, model_config: Dict[str, Any]) -> None:
+        # import in class level to avoid installing nougat package
+        from torch.utils.data import DataLoader, ConcatDataset
+        self.DataLoader =  DataLoader
+        self.ConcatDataset = ConcatDataset
+        try:
+            from nougat import NougatModel
+            from nougat.utils.dataset import LazyDataset
+            from nougat.utils.device import move_to_device
+            from nougat.utils.checkpoint import get_checkpoint
+            from nougat.postprocessing import markdown_compatible
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError(
+                "Please install nougat to use NougatModelServer. You can use `pip install nougat-ocr` to install it."
+            )
+
+        super().__init__(model_config)
+        self._model_config = NougatModelConfig(**self._model_config)
+        checkpoint = get_checkpoint(None, model_tag=self._model_config.model_name)
+        self.model = NougatModel.from_pretrained(checkpoint)
+        self.model = move_to_device(self.model, bf16=False, cuda=self._model_config.batch_size > 0)
+        self.model.eval()
+        self.LazyDataset = LazyDataset
```

Comment:
nit: do we need to hold `LazyDataset` in member variable `self.LazyDataset`. Can we directly use it instantiate dataset class in the __call__ method.